src/analysis/mesh/qgsmeshcontours.h

```diff
@@ -57,21 +57,53 @@ class ANALYSIS_EXPORT QgsMeshContours
      * \param layer mesh layer to be associated with this exporter
      */
     QgsMeshContours( QgsMeshLayer *layer );
+
+    /**
+     * Constructs the mesh contours exporter directly with triangular mesh and dataset values to populate the cache with them.
+     * The cache can't be changed.
+     * This instance can be used in a thread safe way.
+     * \param triangularMesh triangular mesh of the meshlayer
+     * \param nativeMesh the native mesh of the mesh layer
+     * \param datasetValues the dataset values to used for creating contours
+     * \param scalarActiveFaceFlagValues active face flag values
+     *
+     * \since QGIS 3.18
+     */
+    QgsMeshContours( const QgsTriangularMesh &triangularMesh,
+                     const QgsMesh &nativeMesh,
+                     const QVector<double> &datasetValues,
+                     const QgsMeshDataBlock scalarActiveFaceFlagValues ) SIP_SKIP;
+
+
     ~QgsMeshContours();
 
     /**
-     * Exports multi line string containing the contour line for particular dataset and value
-     * \param index dataset index
+     * Exports multi line string containing the contour line for particular dataset and value.
+     * \param index dataset index that is used to update the cache
      * \param value value of the contour line
      * \param method for datasets defined on faces, the method will be used to convert data to vertices
      * \param feedback optional feedback object for progress and cancellation support
      * \returns MultiLineString geometry containing contour lines
+     *
+     * \note not thread safe
      */
     QgsGeometry exportLines( const QgsMeshDatasetIndex &index,
```

Comment:
I think all these not thread safe methods should be marked as deprecated

Reply:
new method are not exposed to python